src/rust/engine/fs/store/src/remote.rs

```diff
@@ -41,37 +46,65 @@ impl ByteStore {
     instance_name: Option<String>,
     root_ca_certs: Option<Vec<u8>>,
     oauth_bearer_token: Option<String>,
-    thread_count: usize,
+    _thread_count: usize,
     chunk_size_bytes: usize,
     upload_timeout: Duration,
-    backoff_config: BackoffConfig,
+    _backoff_config: BackoffConfig,
     rpc_retries: usize,
-    connection_limit: usize,
+    _connection_limit: usize,
   ) -> Result<ByteStore, String> {
-    let env = Arc::new(grpcio::Environment::new(thread_count));
-    let env2 = env.clone();
-
-    let connect = move |cas_address: &str| {
-      let builder = grpcio::ChannelBuilder::new(env2.clone());
-      if let Some(ref root_ca_certs) = root_ca_certs {
-        let creds = grpcio::ChannelCredentialsBuilder::new()
-          .root_cert(root_ca_certs.clone())
-          .build();
-        builder.secure_connect(cas_address, creds)
-      } else {
-        builder.connect(cas_address)
+    let tls_client_config = match root_ca_certs {
+      Some(pem_bytes) => {
+        let mut tls_config = ClientConfig::new();
+        let mut reader = std::io::Cursor::new(pem_bytes);
+        tls_config
+          .root_store
+          .add_pem_file(&mut reader)
+          .map_err(|_| "unexpected state in PEM file add".to_owned())?;
+        Some(tls_config)
       }
+      _ => None,
+    };
+
+    let scheme = if tls_client_config.is_some() {
+      "https"
+    } else {
+      "http"
     };
 
-    let serverset = Serverset::new(cas_addresses, connect, connection_limit, backoff_config)?;
```

Comment:
I removed use of `Serverset` here for now. Needs research on how this would interact with Tonic's own management of dynamic endpoints.

cc @illicitonion @stuhood